magento/inventory
# MSI-2724: Inventory is not updated when Product SKU is changed.

### Description (*)


### Fixed Issues (if relevant)

1. magento/inventory#2724: Inventory is not updated when Product SKU is changed
2. https://github.com/magento/inventory/issues/574: Changes in Product SKU
3. https://github.com/magento/inventory/issues/2001: Quantity Incorrect After Changing SKU
4. https://github.com/magento/inventory/issues/2714: Export stock sources: exports old products, that are no longer exists
5. https://github.com/magento/inventory/issues/2773: Credit memo: Salable Quantity Greater than Quantity

### Manual testing scenarios (*)

Note: Enable 'Synchronize Catalog with Inventory' in Admin -> Configuration -> Catalog -> Inventory -> Stock Options.

1. Create simple product with default source and custom source assigned.
2. Change simple product sku via admin.
3. Verify "inventory_source_item" table and "inventory_low_stock_notification_configuration". Source items and configurations for default and custom source should be updated.
4. Change simple product sku via WebApi.
POST "rest/V1/products" with body {"product" : {"id" : product_id, "sku" : "new_product_sku"}}
5. Verify "inventory_source_item" table and "inventory_low_stock_notification_configuration". Source items and configurations for default and custom source should be updated.
6. Delete simple product.
7. Verify "inventory_source_item" table and "inventory_low_stock_notification_configuration". Source items and configurations for default and custom source should be deleted.
8. Create simple product again and delete it with WebApi.
9. Verify inventory_source_item table. Source items for default and custom source should be deleted.
10. Create simple product once again and delete it with import.
11. Verify "inventory_source_item" table and "inventory_low_stock_notification_configuration". Source items and configurations for default and custom source should be deleted.

### Questions or comments


### Related PRs
Merchdocs: https://github.com/magento/merchdocs/pull/366

### Contribution checklist (*)
